sunpy/map/mapbase.py

```diff
@@ -2501,6 +2502,68 @@ def _check_axes(self, axes, warn_different_wcs=False):
 
         return axes
 
+    def reproject_to(self, output_projection, algorithm='interp', **reproject_args):
+        """
+        Reproject the map
+
+        ..note::
+            This method requires the optional package `reproject` to be installed.
+
+        Parameters
+        ----------
+        output_projection : `dict` or `~astropy.wcs.WCS`
+            The destination FITS WCS header or WCS instance
+        algorithm : str
+            One of the supported `reproject` algorithms (see below)
+
+        Returns
+        -------
+        outmap : `~sunpy.map.GenericMap`
+            The reprojected map
+
+        Notes
+        -----
+        The supported `reproject` algorithms are:
+        * 'interp' for :func:`~reproject.reproject_interp`
+        * 'adaptive' for :func:`~reproject.reproject_adaptive`
+        * 'exact' for :func:`~reproject.reproject_exact`
+        See the respective documentation for these functions for additional keyword
+        arguments that are allowed.
+
+        If the output projection is specified by a `~astropy.wcs.WCS` instance, the
+        `shape_out` keyword argument must also be specified for the dimensions of the
+        output map.  The ordering of these dimensions is in NumPy order (i.e.,
+        secondary axis and then primary axis).
+        """
+        try:
+            import reproject
+        except ImportError:
+            raise ImportError("This method requires the optional package `reproject`.")
+
+        if not isinstance(output_projection, astropy.wcs.WCS):
+            output_projection = Header(output_projection)
+
+        # Select the desired reprojection algorithm
+        functions = {'interp': reproject.reproject_interp,
+                     'adaptive': reproject.reproject_adaptive,
+                     'exact': reproject.reproject_exact}
+        if algorithm not in functions:
+            raise ValueError(f"The specified algorithm must be one of: {list(functions.keys())}")
+        func = functions[algorithm]
+
+        # Block the `return_footpoint` keyword argument from being passed on
+        if 'return_footprint' in reproject_args:
+            del reproject_args['return_footprint']
```

Comment:
```suggestion
        reproject_args.pop('return_footprint', None)
```
is one less line?

Reply:
I know that what you suggest is a code pattern that some use, but I don't like it.  I have a philosophical objection to retrieving a value with no intention of using it.

Regardless, it's moot.  I plan to add a message alerting the user when the `return_footprint` keyword is quashed, and so the `if...del` code pattern has to be used.

Reply:
Heh, I came up with a cleaner approach.  Why remove it at all if I'm always going to add it back in later?